apps/backend/utils/unified_ai_client.py

```
@@ -380,7 +419,16 @@ def embed(self, provider: str, text: str, model: Optional[str] = None) -> Dict[s
             raise ValueError(f"Provider '{provider}' is not supported or configured.")
 
-        return strategy.embed(text, model)
+        return await strategy.embed(text, model)
+
+    async def shutdown(self):
+        """Shuts down all provider strategies and closes their connections."""
+        logger.info("🔌 Shutting down UnifiedAIClient strategies...")
+        for name, strategy in self._strategies.items():
+            try:
+                await strategy.close()
+            except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)